### PR TITLE
Write files consistent

### PR DESCRIPTION
**Pull request by khenderick**

### CODE REVIEW

File: ovs/extensions/generic/sshclient.py [hunk 1]
```diff
@@ -523,20 +523,27 @@ def file_write(self, filename, contents, mode='w'):
         :param contents: Contents to write to the file
         :param mode: Mode to write to the file, can be a, a+, w, w+
         """
+        temp_filename = '{0}~'.format(filename)
         if self.is_local is True:
-            with open(filename, mode) as the_file:
+            with open(temp_filename, mode) as the_file:
```
Comment (domsj) [inline ovs/extensions/generic/sshclient.py]: what happens if a bigger temp_file_name file still exists when this operation is started?

Comment (khenderick) [reply ovs/extensions/generic/sshclient.py]: By default (unless the `mode` is overwritten) the file will be truncated if it already exists.

Comment (khenderick) [reply ovs/extensions/generic/sshclient.py]: As usual, I used `strace` to verify whether everything was executed as expected.

Comment (toolslive) [reply ovs/extensions/generic/sshclient.py]: so the caller can provide bad input for mode ?
What happens if mode `"a+"` is used ?

Comment (khenderick) [reply ovs/extensions/generic/sshclient.py]: True, other modes don't make a lot of sense. Removed the option to override it.

File: ovs/extensions/generic/system.py [hunk 2]
```diff
@@ -173,54 +173,3 @@ def get_free_ports(selected_range, exclude=None, nr=1, client=None):
...
                 return free_ports
         raise ValueError('Unable to find requested nr of free ports')
-
-    @staticmethod
-    def read_config(filename, client=None):
-        """
-        Read a configuration file
-        :param filename: File to read
-        :type filename: str
-
```
Comment (JeffreyDevloo) [inline ovs/extensions/generic/system.py:183]: Removed because deprecated?

File: ovs/extensions/generic/system.py [hunk 3]
```diff
@@ -173,54 +173,3 @@ def get_free_ports(selected_range, exclude=None, nr=1, client=None):
...
-            cp.readfp(StringIO(contents))
-            return cp
-
-    @staticmethod
-    def write_config(config, filename, client=None):
-        """
-        Write a configuration file
-        :param config: Contents to write
-        :type config: RawConfigParser
-
```
Comment (JeffreyDevloo) [inline ovs/extensions/generic/system.py:208]: Removed because deprecated?

**Review by JeffreyDevloo — CHANGES_REQUESTED** [review]

File: ovs/extensions/generic/system.py (hunk 2 above)
Comment (khenderick) [reply ovs/extensions/generic/system.py:183]: Yes

File: ovs/extensions/generic/system.py (hunk 3 above)
Comment (khenderick) [reply ovs/extensions/generic/system.py:208]: No, just didn't like them, they are used like _everywhere_

**Review by JeffreyDevloo — APPROVED** [review]

**Pull request merged**